Spigot-Server-Patches/0406-Add-Zstd-LZ4-GZIP-write-support-for-region-files.patch

```diff
@@ -0,0 +1,1123 @@
+         </dependency>
++        <!-- Paper - Zstd/LZ4 for region files -->
++        <dependency>
++            <groupId>com.github.luben</groupId>
++            <artifactId>zstd-jni</artifactId>
++            <version>1.4.2-1</version>
++        </dependency>
++        <dependency>
++            <groupId>org.lz4</groupId>
++            <artifactId>lz4-java</artifactId>
++            <version>1.6.0</version>
++        </dependency>
+         <!-- testing -->
+         <dependency>
+             <groupId>junit</groupId>
+diff --git a/src/main/java/com/destroystokyo/paper/PaperConfig.java b/src/main/java/com/destroystokyo/paper/PaperConfig.java
+index 5942c3438..117ffd4ff 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperConfig.java
+@@ -388,4 +388,32 @@ public class PaperConfig {
+         maxBookPageSize = getInt("settings.book-size.page-max", maxBookPageSize);
+         maxBookTotalSizeMultiplier = getDouble("settings.book-size.total-multiplier", maxBookTotalSizeMultiplier);
+     }
++
++    public static int compressionMethod = 2;
++    private static void compressionMethod() {
++        compressionMethod = getInt("settings.compression-method", 2);
++        if (compressionMethod < 0 || compressionMethod > 127 || (compressionMethod > 2 && compressionMethod < 126)) {
++            fatal("compression-method must be 0, 1, 2, 126, or 127");
++        }
++
++        if (compressionMethod == 127) {
++            log("Using Zstd compression.");
++        } else if (PaperConfig.compressionMethod == 126) {
++            log("Using LZ4 compression.");
++        } else if (PaperConfig.compressionMethod == 2) {
++            log("Using Zlib compression.");
++        } else if (PaperConfig.compressionMethod == 0) {
++            log("Using NOP (no) compression.");
++        } else {
++            log("Using GZIP compression.");
++        }
++    }
++
++    public static boolean zstdUseHighCompression = false;
++    private static void zstdUseHighCompression() {
++        zstdUseHighCompression = getBoolean("settings.zstd-use-high-compression", false);
++        if (compressionMethod == 127 && zstdUseHighCompression) {
++            log("Using high compression level (make my PC hurt).");
++        }
++    }
+ }
+diff --git a/src/main/java/net/minecraft/server/RegionFile.java b/src/main/java/net/minecraft/server/RegionFile.java
+index 41f1e15cb..7f55e1b42 100644
+--- a/src/main/java/net/minecraft/server/RegionFile.java
++++ b/src/main/java/net/minecraft/server/RegionFile.java
+@@ -1,5 +1,6 @@
+ package net.minecraft.server;
+ 
++import com.destroystokyo.paper.PaperConfig;
+ import com.destroystokyo.paper.exception.ServerInternalException;
+ import com.google.common.collect.Lists;
+ import java.io.BufferedInputStream;
+@@ -12,6 +13,7 @@ import java.io.File;
+ import java.io.IOException;
+ import java.io.RandomAccessFile;
+ import java.util.List;
++import java.util.zip.DataFormatException;
```

Comment:
needs //paper comment or removal, same for the paperconfig import

Reply:
Totally didn't accidentally leave in imports. Nope.